lib/ex_unit/test/ex_unit_test.exs

```diff
@@ -498,6 +498,77 @@ defmodule ExUnitTest do
     ExUnit.configure(seed: global_seed)
   end
 
+  @doc """
+    Skipped and excluded tests should be included in the stats as well as printed to stdout.
+    On the other hand, invalid tests should be marked as failures in the stats,
+    but still be printed as "invalid" to stdout.
+
+    If setup_all fails, the skipped and excluded tests should not be counted as invalid or failures.
+  """
+  test "skipped and excluded" do
+    defmodule SkippedAndExcludedTests do
+      use ExUnit.Case
+
+      setup_all do
+        raise "oops"
+      end
+
+      @tag :skip
+      test "skipped", do: assert(false)
+
+      test "pass #{__ENV__.line}", do: assert(true)
+      test "pass #{__ENV__.line}", do: assert(true)
+      test "fail #{__ENV__.line}", do: assert(false)
+      test "fail #{__ENV__.line}", do: assert(false)
+
+      @tag :exclude
+      test "exclude me please", do: assert(false)
+    end
+
+    ExUnit.Server.modules_loaded()
+    on_exit_reload_config()
+
+    output =
+      capture_io(fn ->
+        predictable_ex_unit_start(exclude: [:exclude])
+        assert ExUnit.run() == %{total: 6, failures: 4, excluded: 1, skipped: 1}
+      end)
+
+    refute output =~ max_failures_reached_msg()
+    assert strip(output) =~ "6 tests, 0 failures, 1 excluded, 4 invalid, 1 skipped"
+  end
+
+  test "check for invalid and excluded tests in stdout" do
```

Comment:
What is this test doing  that is different from the precious one?

Reply:
sorry, it is an accident duplicate